spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1594,6 +1598,34 @@ public void onMessage(Message message) {
 		}
 	}
 
+	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+		channel.basicQos(1);
+		final CountDownLatch latch = new CountDownLatch(1);
+		QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+			@Override
+			public void handleCancel(String consumerTag) throws IOException {
+				super.handleCancel(consumerTag);
+				latch.countDown();
+			}
+
+			@Override
+			public void handleConsumeOk(String consumerTag) {
+				super.handleConsumeOk(consumerTag);
+				latch.countDown();
+			}
+
+		};
+		channel.basicConsume(queueName, consumer);
+		if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
Sorry, I don't understand; this code is before the nextDelivery() call and is there to force the channel closed if we don't properly set up the consumer.